docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc

```diff
@@ -0,0 +1,266 @@
+[[get-snapshot-status-api]]
+=== Get snapshot status API
+++++
+<titleabbrev>Get snapshot status</titleabbrev>
+++++
+
+Retrieves a detailed description of the current state for each shard participating in the snapshot.
+
+////
+[source,console]
+----
+PUT /_snapshot/my_repository
+{
+  "type": "fs",
+  "settings": {
+    "location": "my_backup_location"
+  }
+}
+
+PUT /_snapshot/my_repository/my_snapshot?wait_for_completion=true
+
+PUT /_snapshot/my_repository/snapshot_2?wait_for_completion=true
+----
+// TESTSETUP
+////
+
+[source,console]
+----
+GET /_snapshot/my_repository/my_snapshot/_status
+----
+
+[[get-snapshot-status-api-request]]
+==== {api-request-title}
+
+`GET /_snapshot/<repository>/<snapshot>/_status`
+
+[[get-snapshot-status-api-desc]]
+==== {api-description-title}
+
+Use the get snapshot status API to retrieve detailed information about snapshots currently running in the cluster.
+
+If you specify both the repository name and snapshot, the request retrieves detailed status information for the given snapshot, even if not currently running.
+
+[[get-snapshot-status-api-path-params]]
+==== {api-path-parms-title}
+
+`<repository>`::
+(Required, string)
+Snapshot repository name used to limit the request.
++
+Omitting a repository name retrieves a list of all snapshots currently running in the cluster.
```

Comment:
You also need to omit the `<snapshot>` param, right?

I'd reword this so it's imperative and places the desired info at the beginning:

`To get a list of all snapshots currently running in the cluster, omit this and the `<snapshot>` parameter.

Also: are wildcard expressions supported?

Reply:
To retrieve a list of all snapshots currently running in the cluster, you omit both `repository` and `snapshot`. To get a list of all snapshots running in a repository, you only omit `snapshot`. Also, wildcards are supported for `repository` (in 8.0) but not for `snapshot`. Therefore, I made some changes to both parameters:

```
`<repository>`::
(Required, string)
Snapshot repository name used to limit the request. Wildcard (`*`) expressions are supported.
+
To retrieve a list of all snapshots in the cluster, omit this and the `<snapshot>` parameter.

`<snapshot>`::
(Required, string)
Comma-separated list of snapshot names to retrieve status for.
+
To retrieve a list of all snapshots in a specified repository, omit this parameter.
+
NOTE: Wildcard (`*`) expressions are not supported for `<snapshot>`.
```